develop/parachain-devs/runtime-development/FRAME/create-custom-pallet.md

```diff
@@ -0,0 +1,334 @@
+
+In this step, you'll configure the pallet to emit events. 
+
+Replace the existing #[pallet::config] line with the following code block:
+
+```rust
+#[pallet::config]
+pub trait Config: frame_system::Config {
+    /// The overarching runtime event type.
+    type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+    /// A type representing the weights required by the dispatchables of this pallet.
+    type WeightInfo: WeightInfo;
+}
+```
+
+## Pallet Events
+
+After configuring the pallet to emit events, the next step is to define the events that can be triggered by functions within the pallet. Events provide a straightforward way to inform external entities such as dApps, chain explorers, or users, that a significant change has occurred in the runtime. In a FRAME pallet, the details of each event and its parameters are included in the node’s metadata, making them accessible to external tools and interfaces.
+
+The `generate_deposit` macro generates a `deposit_event` function on the Pallet, which converts the pallet’s event type into the `RuntimeEvent` (as specified in the `Config` trait) and deposits it using `frame_system::Pallet::deposit_event`.
+
+This step adds an event called `SomethingStored`, which is triggered when a user successfully stores a value in the pallet. The event records both the value and the account that performed the action.
+
+To define events, replace the `#[pallet::event]` line with the following code block:
+
+```rust
+#[pallet::event]
+#[pallet::generate_deposit(pub(super) fn deposit_event)]
+pub enum Event<T: Config> {
+    /// A user has successfully set a new value.
+    SomethingStored {
+        /// The new value set.
+        something: u32,
+        /// The account who set the new value.
+        who: T::AccountId,
+    },
+}
+```
+
+## Pallet Errors
+
+While events signal the successful completion of calls, errors indicate when and why a call has failed. It's essential to use informative names for errors to clearly communicate the cause of failure. Like events, error documentation is included in the node's metadata, so providing helpful descriptions is crucial.
+
+Errors are defined as an enum named `Error` with a generic type. Variants can have fields or be fieldless. Any field type specified in the error must implement the `scale_info::TypeInfo` trait, and the encoded size of each field should be as small as possible. Runtime errors can be up to 4 bytes in size, allowing the return of additional information when needed.
+
+This step defines two basic errors: one for handling cases where no value has been set, and another for managing arithmetic overflow.
```

Comment:
This is nitpicking so it's up to you, but I'd move this sentence AFTER the code snippet